tests/e2e/reconnect.spec.ts

```diff
@@ -0,0 +1,230 @@
+    expect(jwks.keys.length).toBeGreaterThan(0);
+    const key = jwks.keys.find((key) => key.kid === keyId);
+    expect(key).toMatchObject({ kid: keyId });
+
+    return { accountId, walletId, keyId };
+  });
+
+  await test.step('shows connect consent page', async () => {
+    await page.waitForURL((url) =>
+      url.pathname.startsWith('/grant-interactions'),
+    );
+    await waitForGrantConsentPage(page);
+  });
+
+  await test.step('connects', async () => {
+    const continueWaitMs = await continueWaitMsPromise;
+    await acceptGrant(page, continueWaitMs);
+    await waitForWelcomePage(page);
+
+    await expect(background).toHaveStorage({ connected: true });
+  });
+
+  const playgroundUrl = 'https://webmonetization.org/play/';
+  await page.goto(playgroundUrl);
+
+  const monetizationCallback = spy<[Event], void>();
+  await page.exposeFunction('monetizationCallback', monetizationCallback);
+  await page.evaluate(() => {
+    window.addEventListener('monetization', monetizationCallback);
+  });
+
+  await page
+    .getByLabel('Wallet address/Payment pointer')
+    .fill(walletAddressUrl);
+  await page.getByRole('button', { name: 'Add monetization link' }).click();
+
+  await expect(page.locator('link[rel=monetization]')).toHaveAttribute(
+    'href',
+    walletAddressUrl,
+  );
+
+  await page.waitForSelector('#link-events .log-header');
+  await page.waitForSelector('#link-events ul.events li');
+  await expect(page.locator('#link-events ul.events li').last()).toContainText(
+    'Load Event',
+  );
+
+  await test.step('revoke key', async () => {
+    await revokeKey(page, revokeInfo);
+
+    const { keys } = await getJWKS(walletAddressUrl);
+    expect(keys.find((key) => key.kid === revokeInfo.keyId)).toBeUndefined();
+  });
+
+  await test.step('make one-time payment', async () => {
+    await page.goto(playgroundUrl);
+
+    await popup.reload({ waitUntil: 'networkidle' });
+    await page.bringToFront();
+    await popup.waitForSelector(`[data-testid="home-page"]`);
+
+    await expect(popup.getByRole('button', { name: 'Send now' })).toBeVisible();
+    expect(await popup.getByRole('textbox').all()).toHaveLength(1);
+
+    await popup.getByRole('textbox').fill('1.5');
+    await popup.getByRole('button', { name: 'Send now' }).click();
+
+    // payment should not go through
+    await expect(page.locator('link[rel=monetization]')).toHaveAttribute(
+      'href',
+      walletAddressUrl,
+    );
+
+    await page.waitForSelector('#link-events .log-header');
+    await page.waitForSelector('#link-events ul.events li');
+    await expect(
+      page.locator('#link-events ul.events li').last(),
+    ).toContainText('Load Event');
+
+    await expect(monetizationCallback).toHaveBeenCalledTimes(0);
+  });
+
+  await test.step('asks for key-add consent', async () => {
+    // revoked key should be detected
+    await expect(
+      popup.getByRole('button', {
+        name: 'It was a mistake, I’d like to reconnect my wallet',
```

Comment:
```suggestion
        name: i18n.getMessage('keyRevoked_action_reconnect'),
```

(`i18n` is fixture like `popup`)